spotbugs/src/main/java/edu/umd/cs/findbugs/util/MutableClasses.java

```diff
@@ -126,7 +126,8 @@ private boolean computeMutable() {
             }
 
             for (Method method : cls.getMethods()) {
-                if (!method.isStatic() && looksLikeASetter(method)) {
+                if (!method.isStatic() && (!method.isPrivate() || cls.isNested())
```

Comment:
`isNested()` is not sufficient, as private methods are accessible also from the host, hence this check is not sufficient.

If I understand correctly, this heuristic is used to examine the class from the user's perspective.

If that is really the case, the user does not see intra-nest organization nor private methods and therefore we should not concern ourselves with private details.

Reply:
Sorry, I do not fully understand you. In this logical expression we are trying to determine whether the method is a setter. If it is static, then it is no setter. If it is private, it is no setter, except if the class is nested, because as you write, private methods are acessible from the host. Thus in case of nested classes private is not enough to say that this method is not a setter.

Reply:
My question is whether the check catches all everything you expect it to in this scenario:

```
public class Outer {
    private class Inner {
        private boolean fresh;
        
        private void clear() {
            fresh = true;
            cleared(this);
        }
    }

    private List<Inner> inners;
    private int counter;
    
    private void clear() {
        inners.forEach(Inner::clear);
    }
    
    private void cleared(Inner inner) {
        counter++;
    }
}
```

I have a separate argument that we should just ignore all private methods, because it is so easy to hide this mutability from us:
```
public class Outer {
    private class Inner {
        private boolean fresh;
    }

    private List<Inner> inners;
    private int counter;
    
    private void clear() {
        inners.forEach(inner -> {
            inner.fresh = true;
            counter++;
        });
    }
}
```